at_secondary/at_secondary_server/lib/src/connection/outbound/outbound_message_listener.dart

```diff
@@ -80,17 +81,22 @@ class OutboundMessageListener {
           // Right now, all callers of this method only expect there ever to be a 'data:' response.
           // So right now, the right thing to do here is to throw an exception.
           // We can leave the connection open since an 'error:' response indicates normal functioning on the other end
-          throw AtConnectException("Request to remote secondary ${outboundClient.toAtSign} at ${outboundClient.toHost}:${outboundClient.toPort} received error response '$result'");
+          result = result.toString().replaceAll('error:', '');
+          var errorCode = result.toString().split('-')[0];
+          var errorDescription = result.toString().split('-')[1];
+          throw AtExceptionUtils.get(errorCode)..message = errorDescription;
```

Comment:
since we are assigning error description after getting the exact exception object.. we can throw
throw AtExceptionUtils.get(errorCode, errorDescription)..inside AtExceptionUtils create exception instance by passing errorDescription to constructor

Reply:
Modified the get method to take errorCode and errorDescription.